src/components/gridListItem/index.tsx

```diff
@@ -213,7 +213,12 @@ class GridListItem extends Component<GridListItemProps> {
         accessible={renderCustomItem ? true : undefined}
         {...Modifiers.extractAccessibilityProps(this.props)}
       >
-        {imageProps && <Image style={itemSize} {...imageProps} customOverlayContent={children}/>}
+        {imageProps && (
+          <View style={[{borderRadius: imageProps?.borderRadius}, itemSize]}>
+            <Image {...imageProps} style={[itemSize, imageProps?.style]}/>
```

Comment:
I found that the issue is coming from the style override.
```
style={itemSize} {...imageProps}
```
So when the user pass some style via imageProps the size getting override.

I think the better solution will be to fix the style override.

```
{imageProps && <Image {...imageProps} style={[itemSize, imageProps?.style]} customOverlayContent={children}/>}
```

Reply:
So create a PR, if it works I'll close this one

Reply:
[related fix](https://github.com/wix/react-native-ui-lib/pull/3222)